atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1812,6 +1813,17 @@ private void waitForCommitToComplete(Iterable<Long> startTimestamps) {
         return result;
     }
 
+    private Map<Long, Long> loadCommitTimestamps(Set<Long> startTimestamps) {
+        // distinguish between a single timestamp and a batch, for more granular metrics
+        if (startTimestamps.size() == 1) {
+            long singleTs = startTimestamps.iterator().next();
+            Long commitTsOrNull = defaultTransactionService.get(singleTs);
+            return commitTsOrNull == null ? ImmutableMap.of() : ImmutableMap.of(singleTs, commitTsOrNull);
```

Comment:
I guess it's not really feasible to have a test that ensures you didn't change any behaviour. Are there at least tests on `TransactionService.get` to verify that `get(foo)` where `foo` has no matching commit timestamp returns `EMPTY_MAP` rather than `<foo, null>`?

Reply:
Unfortunately `TransactionService` has no tests, but I'm fairly confident that our transaction tests cover this codepath. This behavior is also document in the `KeyValueService#get` interface